src/PostgREST/Query/SqlFragment.hs

```diff
@@ -173,6 +174,12 @@ asJsonSingleF returnsScalar
   | returnsScalar = "coalesce((json_agg(_postgrest_t.pgrst_scalar)->0)::text, 'null')"
   | otherwise     = "coalesce((json_agg(_postgrest_t)->0)::text, 'null')"
 
+asXmlSingleF :: Bool -> SqlFragment
+asXmlSingleF returnsScalar
+  | returnsScalar = "coalesce(xmlagg(_postgrest_t.pgrst_scalar)::text, '')"
+  -- RPC not returning scalar: Hope every column in the table is xml and concat everything:
+  | otherwise     = "coalesce(string_agg((SELECT string_agg(json_each_text.value, '') FROM json_each_text(row_to_json(_postgrest_t))), '\n'), '')"
```

Comment:
Interesting :eyes:! Do you have a sample output of this query?

Btw, the codecov check will get green once you add tests.

Reply:
The support of scalar xml return type is surely helpful. Whether the support of table return type is helpful, I am not sure. Nevertheless, the latter works like that:

The `RETURNS TABLE` function:
```sql
create or replace function public.x_table() returns table(xml xml, xml2 xml) language sql as $$
	select '<xml1/>'::xml, '<b1/>'::xml
	union all
	select '<xml2/>'::xml, '<b2/>'::xml
$$
```

The output on `curl --header 'Accept: text/xml' localhost:3000/rpc/x_table`:
```
<xml1/><b1/>
<xml2/><b2/>
```
Synergies with #1582 expected.

Reply:
Hm, with a composite return type the column names should not be lost. How about something based on `xml_forest`?

https://stackoverflow.com/a/65245107

Reply:
Interesting idea. I can't manage it to get to work: Both xmlforest and xmlelement seem to need hardcoded parameter list resp. element names, thus would need some dynamic sql. How could that be integrated into the haskell code?

Meanwhile, the poor man's solution would be text-concatenation:

```sql
coalesce(string_agg((SELECT string_agg('<' || json_each_text.key || '>' || json_each_text.value || '</' || json_each_text.key || '>', '') FROM json_each_text(row_to_json(pg_user))), '\n'), '')
```

The json_each_text.key and json_each_text.value should probably be xml-escaped?

Reply:
Note to myself: Above code will give undesired result if json_each_text.value is null.

Reply:
I thought a bit more about this and I think the basic question is: What kind of level do we want to have for XML support?

We support three levels of "content formatting" or whatever we call it:
- JSON does format multiple columns and also embedded resources into one big proper JSON document
- CSV formats the top-level nicely, but all nested resources pose a challenge. However, this seems to be a limitation of the CSV format itself, not strictly our implementation.
- TEXT/BYTEA formats only a single column - but does not support multiple columns in a useful way.

Even if we can make the multi-column XML approach work like suggested in this comment thread, it will still only be similar to the CSV style, right? So embedded resources would not be properly formatted as XML etc.

Since XML does in fact allow the nesting there is no point in implementing a half-baked solution like CSV does. However to properly implement a full XML representation including columns and nested resources, considerably more work would need to be done, correct?

I would suggest to focus this PR on only implementing XML support similar to TEXT/BYTEA, because everything else seems quite complex.

This means it could be enough to just add the xml media type here:

https://github.com/PostgREST/postgrest/blob/f3f9030c4897e93e50e6f5dcecb317f2dca31a1b/src/PostgREST/App.hs#L620-L622

And then treat it as a binary field. Not sure if that works with the built-in casting, but you can try.

Reply:
> I would suggest to focus this PR on only implementing XML support similar to TEXT/BYTEA

I agree, as I said previously,

> Whether the support of table return type is helpful, I am not sure.

Reply:
> This means it could be enough to just add the xml media type here [...] And then treat it as a binary field.

This still fails with Postgres error "Function string_agg(xml, unknown) does not exist" as reported in #2188.  I guess this happens in `SqlFragment.asBinaryF`?

So I suppose we need to fiddle "the information that the response shall be xml" though the call stack, from `App.handleInvoke` to `Statements.callProcStatement`, and add a function `SqlFragment.asXmlF`, analogous to `asBinaryF`?

See my updated PR, thank you.